tools/oshinko-deploy.sh

```diff
@@ -55,71 +82,54 @@ while getopts :s:w:p:u:h opt; do
     esac
 done
 
-if [ -z "$PROJECT" ]
-then
-    echo "project name not set, using default value"
-    PROJECT=$DEFAULT_OPENSHIFT_PROJECT
-fi
-
 if [ -z "$OS_USER" ]
 then
     echo "user not set, using default value"
     OS_USER=$DEFAULT_OPENSHIFT_USER
 fi
 
+if [ -z "$PROJECT" ]
+then
+    echo "project name not set, using default value"
+    PROJECT=$DEFAULT_OPENSHIFT_PROJECT
+fi
+
 if [ -z "$SPARK_IMAGE" ]
 then
     SPARK_IMAGE=$DEFAULT_SPARK_IMAGE
 fi
 
-oc login -u system:admin
-oc project default
-REGISTRY=$(oc get service docker-registry --no-headers=true | awk -F ' ' '{print $2":"$4}' | sed "s,/TCP$,,")
+if [ -z "$WEB_IMAGE" ]
+then
+    WEB_IMAGE=$DEFAULT_OSHINKO_WEB_IMAGE
+fi
 
-# reset back to the default development account
-oc login -u $OS_USER
-oc project $PROJECT
+if [ -z "$REST_IMAGE" ]
+then
+    REST_IMAGE=$DEFAULT_OSHINKO_REST_IMAGE
+fi
 
-# Wait for the registry to be fully up
-r=1
-while [ $r -ne 0 ]; do
-    docker login -u $(oc whoami) -e "jack@jack.com" -p $(oc whoami -t) $REGISTRY
-    r=$?
-    sleep 1
-done
+if [ -z "$WEBROUTE" ]
+then
+    WEBROUTE=$DEFAULT_WEBROUTE
+fi
 
-docker tag oshinko-rest $REGISTRY/$PROJECT/oshinko-rest
-docker push $REGISTRY/$PROJECT/oshinko-rest
-docker tag oshinko-webui $REGISTRY/$PROJECT/oshinko-webui
-docker push $REGISTRY/$PROJECT/oshinko-webui
-docker tag radanalytics-pyspark $REGISTRY/$PROJECT/radanalytics-pyspark
-docker push $REGISTRY/$PROJECT/radanalytics-pyspark
-
-# check to see if we have a local copy of the spark image
-# otherwise, pull it before tagging for oshinko
-local_spark_image=$(docker images -q $SPARK_IMAGE)
-if [ -z "$local_spark_image" ]
+if $OS_ALLINONE
 then
-    docker pull $SPARK_IMAGE
+    oc cluster up
 fi
-docker tag $SPARK_IMAGE $REGISTRY/$PROJECT/oshinko-spark
-docker push $REGISTRY/$PROJECT/oshinko-spark
 
-# set up the oshinko service account
+oc login $OS_CLUSTER -u $OS_USER
```

Comment:
Are the OS_ALLINONE case and the OS_CLUSTER case mutually exclusive?  If I'm using oc cluster up, then I've got no use for OS_CLUSTER, correct? Should we stop them from using -d and -c at the same time?